token-minter/main.go

```diff
@@ -112,40 +127,45 @@ func mintToken(ctx context.Context, opts options) metav1.Time {
 				return false, fmt.Errorf("invalid kubeconfig: %w", err)
 			}
 			return true, nil
-		}); err != nil {
-			panic(err)
+		}, timeout.Done()); err != nil {
+			return metav1.Time{}, fmt.Errorf("timed out waiting for kubeconfig secret: %w", err)
 		}
 	} else {
 		loadingRules := clientcmd.ClientConfigLoadingRules{ExplicitPath: opts.kubeconfigPath}
 		clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(&loadingRules, &clientcmd.ConfigOverrides{})
 		var err error
 		restConfig, err = clientConfig.ClientConfig()
 		if err != nil {
-			panic(err)
+			return metav1.Time{}, fmt.Errorf("failed to get client config: %w", err)
 		}
 	}
 
 	clientset, err := kubernetes.NewForConfig(restConfig)
 	if err != nil {
-		panic(err)
+		return metav1.Time{}, fmt.Errorf("failed to get guest kube client: %w", err)
 	}
 
 	// Get the service account
 	var serviceAccount *corev1.ServiceAccount
-	err = wait.PollImmediate(time.Second*5, time.Minute*2, func() (done bool, err error) {
+	timeout, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
+	err = wait.PollImmediateUntil(time.Second*5, func() (done bool, err error) {
```

Comment:
not understanding what this does differently vs the previous `PollImmediate`

Reply:
In this version, the parent context passed to the function can interrupt the poll, whereas in the prior version the poll is not interruptible (e.g. from a process signal).